ostree/ostree_transport.go

```diff
@@ -0,0 +1,177 @@
+package ostree
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/containers/image/docker/policyconfiguration"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/types"
+)
+
+// Transport is an ImageTransport for ostree paths.
+var Transport = ostreeTransport{}
+
+type ostreeTransport struct{}
+
+func (t ostreeTransport) Name() string {
+	return "ostree"
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+// (i.e. a valid PolicyConfigurationIdentity() or PolicyConfigurationNamespaces() return value).
+// It is acceptable to allow an invalid value which will never be matched, it can "only" cause user confusion.
+// scope passed to this function will not be "", that value is always allowed.
+func (t ostreeTransport) ValidatePolicyConfigurationScope(scope string) error {
+	if strings.Index(scope, "@") < 0 {
+		return errors.Errorf("Invalid scope %s: Must include a repo", scope)
+	}
+	return nil
+}
+
+// ostreeReference is an ImageReference for ostree paths.
+type ostreeReference struct {
+	ref   string
+	repo  string
+	image reference.Named
+}
+
+func (t ostreeTransport) ParseReference(ref string) (types.ImageReference, error) {
+	var repo = ""
+	var imageStr = ""
+	s := strings.SplitN(ref, "@", 2)
+	if len(s) == 1 {
+		imageStr, repo = s[0], "/ostree/repo"
+	} else {
+		imageStr, repo = s[0], s[1]
+	}
+
+	var imageTagged reference.Named
+	s = strings.SplitN(imageStr, ":", 2)
+
+	image, err := reference.WithName(s[0])
```

Comment:
ParseNormalizedNamed is also adds the registry if it is not specified.  This should not be done for system containers.  If I pass "busybox", I don't want it to be resolved to "docker.io/busybox".  Is there a way to not add the registry when it is missing?